src/goRunTestCodelens.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import vscode = require('vscode');
+import { CodeLensProvider, TextDocument, CancellationToken, CodeLens, Command } from 'vscode';
+import { getTestFunctions } from './goTest';
+
+export class GoRunTestCodeLensProvider implements CodeLensProvider {
+	public provideCodeLenses(document: TextDocument, token: CancellationToken): CodeLens[] | Thenable<CodeLens[]> {
```

Comment:
How about adding a codelens at the package level as well? Every go file has a `package` statement up top. So we can add a code lens there and map it to the `go.test.package` command